python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -928,3 +925,32 @@ def evaluate_for_asset(
                 context.asset_key, context.partitions_def, rate_limited_asset_partitions
             ),
         )
+
+
+@whitelist_for_serdes
+class BlockingAssetCheckRule(AutoMaterializeRule, NamedTuple("_BlockingAssetcheckRule", [])):
+    @property
+    def decision_type(self) -> AutoMaterializeDecisionType:
+        return AutoMaterializeDecisionType.SKIP
+
+    def evaluate_for_asset(
+        self, context: AssetConditionEvaluationContext
+    ) -> "AssetConditionResult":
+        from dagster._core.storage.asset_check_execution_record import (
+            AssetCheckExecutionRecordStatus,
+        )
+
+        parent_assets = context.instance_queryer.asset_graph.get_parents(context.asset_key)
```

Comment:
Feel very free to ignore, but it feels like in most cases that we care about blocking checks, it's going to be from the perspective of the downstream asset looking for upstream blocking checks.

If that's the case, a method on the asset graph to do basically what you have right here might be useful.